dotCMS/src/main/webapp/html/js/dotcms/dijit/image/ImageEditor.js

```diff
@@ -384,57 +373,79 @@ dojo.declare("dotcms.dijit.image.ImageEditor", dijit._Widget,{
 
     saveImage : function(){
         if(this.binaryFieldId != null && this.binaryFieldId.length > 0){
-            this.saveBinaryImage();
+            this.saveBinaryImage(this.activeEditor);
         }
         else{
             this.saveFileImage();
         }
     },
 
 
-
+_isValidURL: function (url) {
+		var elm;
+		if (!elm) {
+			elm = document.createElement("input");
+			elm.setAttribute("type", "url");
+		}
+		elm.value = url;
+		return elm.validity.valid;
+	},
     /**
      * This saves an image
      * that lives on a contentlet
      *
      */
 
-    saveBinaryImage: function(){
+    saveBinaryImage: function(activeEditor){
 
         var field=this.binaryFieldId;
         if(this.fieldContentletId.length>0) {
             field=this.fieldContentletId;
         }
 
-        var url =   this.cleanUrl(this.currentUrl) ;
-        url = (url.indexOf("?")>-1) ? url + "&"  : url + "?";
-        url += (field.length > 0) ? "&binaryFieldId=" +field : "";
-        url += "&_imageToolSaveFile=true";
-        
-        
-        var xhr = new XMLHttpRequest();
-        xhr.onload = (self => {
-            return () => {
-                if (xhr.status == 200) {
-                    var dataJson = JSON.parse(xhr.responseText);
-                    self.tempId=dataJson.id;
-                    if(window.document.getElementById(self.binaryFieldId + "ValueField")){
-                        window.document.getElementById(self.binaryFieldId + "ValueField").value=dataJson.id; 
-                    }
-                    
-                } else {
-                    alert("Error! Upload failed");
-                }
-            };
-        })(this);
-        xhr.open("GET", url, true);
-        xhr.send();
-        
-
+        var url = this.cleanUrl(this.currentUrl);
 
-        this.setThumbnail();
-        
+            if (activeEditor) {
+                var newUrl;
+                if (this._isValidURL(url)) {
+                    newUrl = new URL(url).pathname;
+                } else {
+                    newUrl = url;
+            }
 
+            const asset = `
+                <img 
+                    src="${newUrl}" 
+                    alt="${this.fieldName}"
+                    data-field-name="${this.fieldName}"
+                    data-inode="${this.inode}"
+                    data-identifier="${this.fieldContentletId}"
+                    data-saveas="${this.saveAsFileName}"
+                />`;
+            activeEditor.execCommand("mceReplaceContent", false, asset);
+		} else {
+            url = url.indexOf("?") > -1 ? url + "&" : url + "?";
```

Comment:
Maybe extract all this logic to a function.

Reply:
and maybe some doc for future understanding